internal/cluster/sbd.go

```diff
@@ -0,0 +1,210 @@
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	SBDPath            = "/usr/sbin/sbd"
+	SBDConfigPath      = "/etc/sysconfig/sbd"
+	SBDStatusUnknown   = "unknown"
+	SBDStatusUnhealthy = "unhealthy"
+	SBDStatusHealthy   = "healthy"
+)
+
+type SBD struct {
+	cluster string
+	Devices []*SBDDevice      `mapstructure:"devices,omitempty"`
+	Config  map[string]string `mapstructure:"config,omitempty"`
+}
+
+type SBDDevice struct {
+	sbdPath string
+	Device  string     `mapstructure:"device,omitempty"`
+	Status  string     `mapstructure:"status,omitempty"`
+	Dump    SBDDump    `mapstructure:"dump,omitempty"`
+	List    []*SBDNode `mapstructure:"list,omitempty"`
+}
+
+type SBDDump struct {
+	Header          string `mapstructure:"header,omitempty"`
+	Uuid            string `mapstructure:"uuid,omitempty"`
+	Slots           int    `mapstructure:"slots,omitempty"`
+	SectorSize      int    `mapstructure:"sectorsize,omitempty"`
+	TimeoutWatchdog int    `mapstructure:"timeoutwatchdog,omitempty"`
+	TimeoutAllocate int    `mapstructure:"timeoutallocate,omitempty"`
+	TimeoutLoop     int    `mapstructure:"timeoutloop,omitempty"`
+	TimeoutMsgwait  int    `mapstructure:"timeoutmsgwait,omitempty"`
+}
+
+type SBDNode struct {
+	Id     int    `mapstructure:"id,omitempty"`
+	Name   string `mapstructure:"name,omitempty"`
+	Status string `mapstructure:"status,omitempty"`
+}
+
+var sbdDumpExecCommand = exec.Command
+var sbdListExecCommand = exec.Command
+
+func NewSBD(cluster, sbdPath, sbdConfigPath string) (SBD, error) {
+	var s = SBD{cluster: cluster}
+
+	c, err := getSBDConfig(sbdConfigPath)
+	if err != nil {
+		return s, err
+	} else if _, ok := c["SBD_DEVICE"]; !ok {
+		return s, fmt.Errorf("could not find SBD_DEVICE entry in sbd config file")
+	}
+	s.Config = c
+
+	for _, device := range strings.Split(c["SBD_DEVICE"], ";") {
+		sbdDevice := NewSBDDevice(sbdPath, device)
+		err := sbdDevice.LoadDeviceData()
+		if err != nil {
+			log.Printf("Error getting sbd information: %s", err)
+			continue
+		}
+		s.Devices = append(s.Devices, &sbdDevice)
+	}
+
+	return s, nil
+}
+
+func getSBDConfig(sbdConfigPath string) (map[string]string, error) {
+	configMap := make(map[string]string)
+
+	sbdConfFile, err := os.Open(sbdConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not open sbd config file %s", err)
+	}
+
+	defer sbdConfFile.Close()
+
+	sbdConfigRaw, err := ioutil.ReadAll(sbdConfFile)
+
+	if err != nil {
+		return nil, fmt.Errorf("could not read sbd config file %s", err)
+	}
+
+	// Loop through sbd list output and find for matches
+	r := regexp.MustCompile(`(\S+)\s*=(\S+)`)
```

Comment:
The next regexp should remove the lines starting with a comment:
`^\s*(\w+)\s=\s(\S+)$`